src/components/panel/panel.js

```diff
@@ -1272,18 +1284,26 @@ MdPanelRef.prototype._animateOpen = function() {
  * @private
  */
 MdPanelRef.prototype._animateClose = function() {
+  var onRemoving = this._config['onRemoving'];
+  var onRemovingScope = this._config['scope'] || this._$rootScope;
+
   var animationConfig = this._config['animation'];
   if (!animationConfig) {
     this.removeClass('md-panel-is-showing');
     this.removeClass('_md-panel-shown');
+    var onRemoving = onRemovingScope.$eval(onRemoving);
+    onRemoving && onRemoving.resolve && onRemoving.resolve();
```

Comment:
Why not just return `this._$q(onRemovingScope.$eval(onRemoving)`? That's still a promise, which resolves or rejects to whatever the user specifies.